### ISSUE TITLE
Failed to retrieve access token

### ISSUE THREAD

**Issue by AndreaPro** [rospogrigio/daikin_residential#33]
Hello, I have installed your integration through HACS, when I insert my username and password it says that is impossible to retrieve the access token. This is the output of HomeAssistant.

```
This error originated from a custom integration.

Logger: custom_components.daikin_residential.config_flow
Source: custom_components/daikin_residential/config_flow.py:58
Integration: Daikin Residential Controller (documentation, issues)
First occurred: 16:43:46 (1 occurrences)
Last logged: 16:43:46

Failed to retrieve Access Token: ('Failed to retrieve access token: %s', IATError('Issued in the future'))
```

I am running HA in docker, but the TZ of the container is the same of the server.

I have seen that there was another user with a similar issue but in that case apparently the issue solved after reboot. I have tried doing so but nothing changed

**Comment by rospogrigio** [comment]
Don't know how to help you sincerely... are you sure your account uses the European server? Did you try the tokensaver.py script under tools?

**Comment by AndreaPro** [comment]
Yes, here you can find the [output](https://pastebin.com/nuTkWR38)

It appears that the result is the same, I don't know if the more verbose log can help

**Comment by rospogrigio** [comment]
Looks like the login part is successful but then it fails to retrieve the access token... Don't know why. You may try to remove the `+ 3600000` at line 316 and see if that is causing problems... Otherwise I really don't know what to suggest, sorry. You may want to send me your credentials privately and let me try if I can retrieve the token (maybe there's some different setting we are not considering), then you can change the password... I don't have other ideas, sorry

**Comment by AndreaPro** [comment]
Ok, I am sending you temporary credentials to try to fix the issue on your own. I've tried removing the line from the Python script but the output is the same

**Comment by AndreaPro** [comment]
Ok, I am sending you temporary credentials to try to fix the issue on your own. I've tried removing the line from the Python script but the output is the same

**Comment by jguerre** [comment]
hello rospogrigio. I have the same problem. I run the tokensaver in my pc and generate tokenset.json with my Daikin user mail/password, but in Home Assistant no way ... Is there any workaround?

**Comment by rospogrigio** [comment]
I don't know, I think you'd better ask @AndreaPro how he managed to fix the issue...

**Comment by jguerre** [comment]
Ok! Thanks a lot for all the good job @rospogrigio

**Comment by AndreaPro** [comment]
> > Ok, I am sending you temporary credentials to try to fix the issue on your own. I've tried removing the line from the Python script but the output is the same
> 
> Please @AndreaPro , could you help me to fix this? ( I run the tokensaver in my pc and generate tokenset.json with my Daikin user mail/password, but in Home Assistant no way ... Is there any workaround? thanks a lot in advance). Thanks a lot in advance!

Hello,
I have solved by moving everything to a Pi4 where everything worked without issues, but I think that on the main server the issue was due to chrony that was failing and I suspect that this caused issues with the validation time for the access token.

**Comment by jguerre** [comment]
> > > Ok, I am sending you temporary credentials to try to fix the issue on your own. I've tried removing the line from the Python script but the output is the same
> > 
> > 
> > Please @AndreaPro , could you help me to fix this? ( I run the tokensaver in my pc and generate tokenset.json with my Daikin user mail/password, but in Home Assistant no way ... Is there any workaround? thanks a lot in advance). Thanks a lot in advance!
> 
> Hello, I have solved by moving everything to a Pi4 where everything worked without issues, but I think that on the main server the issue was due to chrony that was failing and I suspect that this caused issues with the validation time for the access token.

Much appreciated. Thanks both for your support ! best
